File: deepseek/2.Estructura de Datos Pila (Stack)/pyhton/pila.py
class Stack:
    def __init__(self):
        self.items = []

    def push(self, item):
        self.items.append(item)

    def pop(self):
        if not self.is_empty():
            return self.items.pop()
        return None

    def is_empty(self):
        return len(self.items) == 0

# Interacción con el usuario
stack = Stack()
while True:
    action = input("¿Desea 'push', 'pop' o 'salir'? ").lower()
    if action == "salir":
        break
    elif action == "push":
        item = input("Ingrese el valor a apilar: ")
        stack.push(item)
    elif action == "pop":
        popped = stack.pop()
        print(f"Elemento desapilado: {popped}" if popped else "La pila está vacía")

File: ChatGPT/2.Estructura de Datos Pila (Stack)/Typescript/pila.ts
class ChatGPTStack<T> {
  private elements: T[] = [];

  push(element: T): void {
    this.elements.push(element);
  }

  pop(): T | undefined {
    return this.elements.pop();
  }

  isEmpty(): boolean {
    return this.elements.length === 0;
  }

  print(): void {
    console.log("Contenido de la pila:", this.elements);
  }
}

// Simulación básica de interacción
const pila = new ChatGPTStack<string>();

pila.push("Elemento 1");
pila.push("Elemento 2");
pila.print();
pila.pop();
pila.print();
console.log("¿Está vacía?", pila.isEmpty());
